rasa/core/agent.py

```diff
@@ -383,8 +383,8 @@ def update_model(
 
         # update domain on all instances
         self.tracker_store.domain = domain
-        if hasattr(self.nlg, "templates"):
-            self.nlg.templates = domain.templates if domain else {}
+        if hasattr(self.nlg, "responses"):
```

Comment:
Users can also load `self.nlg` from custom modules (see [here](https://github.com/RasaHQ/rasa/blob/74612b7d5868d95b58a59b609da3d2faa8488611/rasa/core/nlg/generator.py#L70)). Because of that I think it makes sense to keep support for `if hasattr(self.nlg, "templates")` + add a deprecation warning for it.

Reply:
So you mean we should have both right?

Reply:
yes 👍